src/core/cgroup.c

```diff
@@ -2614,7 +2640,7 @@ static int unit_get_cpu_usage_raw(Unit *u, nsec_t *ret) {
                 _cleanup_free_ char *val = NULL;
                 uint64_t us;
 
-                if ((u->cgroup_realized_mask & CGROUP_MASK_CPU) == 0)
+                if (!cpu_accounting_is_cheap() && (u->cgroup_realized_mask & CGROUP_MASK_CPU) == 0)
                         return -ENODATA;
```

Comment:
let's move this check before the cg_all_unified() check above, in a generic fashion:

```
if (get_cpu_accounting_mask() & ~u->cgroup_realized_mask != 0) 
        return -ENODATA;
```
The above checks whether any of the controllers we need are among the unrealized controllers. This kind of check is pretty nice as it works for all three cases the same way.